pyvips/vimage.py

```diff
@@ -899,6 +1000,230 @@ def toarray(self):
         return [[self(x, y) for x in range(self.width)] 
                 for y in range(self.height)]
 
+
+    # # Following will be relevant if we expose the buffer interface:
+    # @property
+    # def __array_interface__(self):
+    #     '''Return a numpy-standard __array_interface__ dictionary.
+    #    
+    #     This is used by some libraries (PIL, e.g.) to interpret the image as an array.
+    #    
+    #     See https://numpy.org/doc/stable/reference/arrays.interface.html for more info.
+    # 
+    #     '''
+    #   
+    #     if self.bands == 1:
+    #         shape = (self.height, self.width)
+    #     else:
+    #         shape = (self.height, self.width, self.bands)
+    # 
+    #     typestr = FORMAT_TO_TYPESTR[self.format]
+    #     
+    #     interface = {
+    #         'shape': shape,
+    #         'strides': None,
+    #         'descr': [('', typestr)],
+    #         'typestr': typestr,
+    #         'version': 3
+    #     }
+    # 
+    #     return interface
+
+
+    def __array__(self, dtype=None):
+        '''Conversion to a NumPy array.
+        
+        Parameters
+        ----------
```

Comment:
pyvips uses (or is supposed to use) the google docstrings conventions: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
Ok, thanks for the pointer.  I'll adjust the code.